### DISCUSSION TITLE
Problem with 4.0 installation

### DISCUSSION THREAD

**Discussion by dluccajose** [thorsten/phpMyFAQ#3159] [Q&A]
I am trying to install the 4.0 version but i get this error when i try to go to the setup page

`Fatal error: Uncaught TypeError: phpMyFAQ\Configuration::getConfigurationInstance(): Return value must be of type phpMyFAQ\Configuration, null returned in /var/www/faq/src/phpMyFAQ/Configuration.php:66 Stack trace: #0 /var/www/faq/src/phpMyFAQ/Template/PluginTwigExtension.php(31): phpMyFAQ\Configuration::getConfigurationInstance() #1 /var/www/faq/src/phpMyFAQ/Template/TwigWrapper.php(54): phpMyFAQ\Template\PluginTwigExtension->__construct() #2 /var/www/faq/src/phpMyFAQ/Controller/Frontend/SetupController.php(137): phpMyFAQ\Template\TwigWrapper->__construct('/var/www/faq/as...') #3 /var/www/faq/src/phpMyFAQ/Controller/Frontend/SetupController.php(54): phpMyFAQ\Controller\Frontend\SetupController->render('setup/index.twi...', Array) #4 [internal function]: phpMyFAQ\Controller\Frontend\SetupController->index() #5 /var/www/faq/src/phpMyFAQ/Application.php(102): call_user_func_array(Array, Array) #6 /var/www/faq/src/phpMyFAQ/Application.php(47): phpMyFAQ\Application->handleRequest(Object(Symfony\Component\Routing\RouteCollection)) #7 /var/www/faq/setup/index.php(101): phpMyFAQ\Application->run(Object(Symfony\Component\Routing\RouteCollection)) #8 {main} thrown in /var/www/faq/src/phpMyFAQ/Configuration.php on line 66`

I am using the zip version, runing on a laradock container with PHP 8.2

**Comment by thorsten** [comment]
@dluccajose Thanks for testing, which version did you use? Alpha.4?

**Comment by thorsten** [comment]
@dluccajose Please check one of the nightly builds, should be fixed: https://github.com/thorsten/phpMyFAQ/releases

**Reply by dluccajose** [reply]
Thanks for the quick response! i tried the last nightly build and the error got fixed, however i couldn't complete the installation, i got to the final step, click on "Submit" and then the browser throws an error for endless redirection and the url looks like this "http://faq.test/setup/setup/setup/setup/setup/setup/setup/setup/setup/setup/setup/setup/setup/setup/setup/setup/setup/setup/setup/setup/setup/"

**Comment by thorsten** [comment]
I'll check it asap!

**Reply by dluccajose** [reply]
i have been doing some digging myself, i think i have two problems, first i solved the endless redirection changing this line in the file Bootstrap.php

`    $response = new RedirectResponse('./setup/');`
to
`    $response = new RedirectResponse('/setup/');`

after that, i tried again to install but nothing happens when submit the last step, it just loaded again the index setup page. A thing that i note it, if i enter a random url like setup/sdfsdf it still redirects me to the setup/ page, in other words, the request is getting sended to the index.php in root and not the index.php of the setup folder, that make me think that my problem could be with the .htaccess or my nginx config, this is my conf file

`server {

    listen 80;
    listen [::]:80;

    # For https
    # listen 443 ssl;
    # listen [::]:443 ssl ipv6only=on;
    # ssl_certificate /etc/nginx/ssl/default.crt;
    # ssl_certificate_key /etc/nginx/ssl/default.key;

    server_name faq.test;
    root /var/www/faq;
    index index.php index.html index.htm;

    location / {
         try_files $uri $uri/ /index.php$is_args$args;
    }

    location ~ \.php$ {
        try_files $uri /index.php =404;
        fastcgi_pass php-upstream;
        fastcgi_index index.php;
        fastcgi_buffers 16 16k;
        fastcgi_buffer_size 32k;
        fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
        #fixes timeouts
        fastcgi_read_timeout 600;
        include fastcgi_params;
    }

    location ~ /\.ht {
        deny all;
    }

    location /.well-known/acme-challenge/ {
        root /var/www/letsencrypt/;
        log_not_found off;
    }

    error_log /var/log/nginx/app_error.log;
    access_log /var/log/nginx/app_access.log;
}`

**Reply by thorsten** [reply]
I didn't had time to check it, but maybe I forgot to fix the loop with ngnix because I'm using mainly Apache. I'll check it.

**Comment by thorsten** [comment]
I found some issues and fixed them:

- https://github.com/thorsten/phpMyFAQ/commit/c40bb776e65cce8de1b870459951f8bb402066de
- https://github.com/thorsten/phpMyFAQ/commit/d76f8862f057cec6b2aa9fc8a3456e1910fd0c5c

One more thing is missing after the 4th form that the Buttons should be removed after hitting the submit button. I'll add that.

**Comment by thorsten** [comment]
... and last but not least, a visual loading indicator while phpMyFAQ installs the database:

https://github.com/thorsten/phpMyFAQ/commit/6793118553eef889968b2a806ec9ccb9cbd06b32

@dluccajose The next nightly build should be working now.

**Reply by dluccajose** [reply]
Thanks!! with the last nightly build and some nginx configuration file that i found on the repo i was finally able to complete the install.
